api/app/src/routes/user.ts

```diff
@@ -129,28 +156,50 @@ router.delete(
     const cxId = getCxIdOrFail(req);
     const connectedUser = await getConnectedUserOrFail({ id: userId, cxId });
 
-    const providerOAuth2 = providerOAuth2OptionsSchema.safeParse(req.query.provider);
+    if (await revokeUserProviderAccess(connectedUser, req.query.provider)) {
+      return res.sendStatus(status.OK);
+    }
+  })
```

Comment:
Routes should always either:
- return a response with headers sent ("closed");
- call `next`;
- throw an error.

Here, if `revokeUserProviderAccess` doesn't work we'll return and leave the client/caller awaiting until timeout.